### PR TITLE
test/ieee802154_hal: general cleanup and cosmetic fixes

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR is a cleanup and cosmetic fix of `test/ieee802154_hal`, based on some of the comments by @fjmolinas during the review of #16826 

The changes are:
892cfc38c7 test/ieee802154_hal: rename long_addr into ext_addr
ec1d5a73e0 test/ieee802154_hal: use od where possible
069dcc4f3b test/ieee802154_hal: use l2util where possible
6618a5c038 test/ieee802154_hal/Makefile: use event_thread instead of event_thread_highest
fab1b26e79 test/ieee802154_hal/Makefile: add shell_commands
1a9338cd54 test/ieee802154_hal: rename packet to frame

### Testing procedure

This should be really easy to test. Simply make sure that all commands work. Sending and receiving should also still work. Here is a sample output:
```
2021-09-29 11:26:35,128 # main(): This is RIOT! (Version: 2021.10-devel-690-gc76b4-pr/test/hal/l2util)
2021-09-29 11:26:35,129 # Trying to register cc2538_rf.
2021-09-29 11:26:35,130 # Success
2021-09-29 11:26:35,135 # Initialization successful - starting the shell now
> 2021-09-29 11:26:37,392 # Frame received:
2021-09-29 11:26:37,395 # 00000000  61  9C  6A  23  00  4D  43  BC  20  E1  FE  8D  AE  DB  3D  74
2021-09-29 11:26:37,397 # 00000010  65  73  74
2021-09-29 11:26:37,398 # LQI: 238, RSSI: 120
2021-09-29 11:26:37,398 # 
print_addr
2021-09-29 11:26:41,853 # print_addr
2021-09-29 11:26:41,855 # AE:8D:FE:E1:20:BC:43:4D
> txtsnAE:8D:FE:E1:20:BC:43:4D 10
2021-09-29 11:26:52,418 # txtsnd AE:8D:FE:E1:20:BC:43:4D 10
2021-09-29 11:26:52,426 # Transmission succeeded
2021-09-29 11:26:52,426 # 
2021-09-29 11:26:52,427 # 
> txtsn00:04:25:19:18:01:BD:DB 10
2021-09-29 11:27:13,128 # txtsnd 00:04:25:19:18:01:BD:DB 10
2021-09-29 11:27:13,136 # Transmission succeeded
2021-09-29 11:27:13,137 # 
2021-09-29 11:27:13,137 # 
2021-09-29 11:27:13,140 # Received valid ACK with sqn 1
2021-09-29 11:27:13,142 # LQI: 238, RSSI: 103
2021-09-29 11:27:13,143 # 
> 
```

### Issues/PRs references

None so far, but followed @fjmolinas comments on #16826

### CODE REVIEW

File: tests/ieee802154_hal/Makefile [hunk 1]
```diff
@@ -21,14 +21,16 @@ DISABLE_MODULE += auto_init_at86rf2xx auto_init_nrf802154
 
 USEMODULE += od
 USEMODULE += luid
+USEMODULE += l2util
```
Comment (miri64) [inline tests/ieee802154_hal/Makefile:24]: For some reason, this is not included for `remote-revb` in the CI...

Comment (fjmolinas) [reply tests/ieee802154_hal/Makefile:24]: Lets run again? I can't reproduce locally...

Comment (miri64) [reply tests/ieee802154_hal/Makefile:24]: If the test is flaky, don't we maybe want to wait with merging this, _after_ feature freeze? Or is this PR critical for the release?

Comment (fjmolinas) [reply tests/ieee802154_hal/Makefile:24]: Nope its a cleanup, it can wait

**Comment by fjmolinas** [comment]
Failed again, at least its not flaky...

**Comment by fjmolinas** [comment]
> Failed again, at least it's not flaky...

Duh.. it's missing the dependency in the app config. It's actually completely out of sync now. How its it that it did not fail the ci??

**Comment by fjmolinas** [comment]
> > Failed again, at least it's not flaky...
> 
> Duh.. it's missing the dependency in the app config. It's actually completely out of sync now. How its it that it did not fail the ci??

Ahhh remote-revb is the only one being built there.

**Comment by fjmolinas** [comment]
@jia200x can you fix the app config?

**Comment by miri64** [comment]
> > > Failed again, at least it's not flaky...
> > 
> > 
> > Duh.. it's missing the dependency in the app config. It's actually completely out of sync now. How its it that it did not fail the ci??
> 
> Ahhh remote-revb is the only one being built there.

The beauty of effectively having two build systems ^^"

**Comment by jia200x** [comment]
fixed and squashed directly

**Comment by fjmolinas** [comment]
@jia200x seems like L2UTIL is not yet defined in kconfig do you mind adding it?

File: tests/ieee802154_hal/app.config.test [hunk 2]
```diff
@@ -4,11 +4,12 @@ CONFIG_MODULE_OD=y
 CONFIG_MODULE_LUID=y
 CONFIG_MODULE_IEEE802154=y
 CONFIG_MODULE_SHELL=y
+CONFIG_MODULE_SHELL_COMMANDS=y
+CONFIG_MODULE_L2UTIL=y
```
[the remark this answers is not in the document]
Comment (leandrolanzieri) [reply tests/ieee802154_hal/app.config.test:8]: oops, just saw https://github.com/RIOT-OS/RIOT/pull/16913#issuecomment-931246299, sorry

**Comment by jia200x** [comment]
done!

**Comment by fjmolinas** [comment]
@jia200x I squashed in the missing kconfig include, hope that is OK!

**Comment by jia200x** [comment]
Oh, sorry. I cherry-picked the commit from another branch and thought it already had the missing include...

**Comment by jia200x** [comment]
unrelated Murdock issue. Let's wait until the problem is fixed

**Review by fjmolinas — APPROVED** [review]
ACK and go

**Pull request merged**